### DISCUSSION TITLE
After-run mod

### DISCUSSION THREAD

**Discussion by ccorrion** [Romanitho/Winget-AutoUpdate#611] [Ideas]
Howdy,

We're using WAU to upgrade apps and one of the things we're looking at adding is a post-run mod.  We would use this for a couple of things:

- Trigger SCCM hardware inventory, so we could have a centralized view of the versions on everything
- Some kind of json/csv output of what got upgraded the last run.

This seems like a pretty simple change to Winget-upgrade.ps1, and a additional WAU-Mods-Postrun template.  Is this something y'all are interested in seeing commited to the main project?  I'm open to implementation ideas as well.

**Comment by KnifMelti** [comment]
You always have (but that is pre-updating):
**Mods for WAU (if Network is active/any Winget is installed/running as SYSTEM):**
https://github.com/Romanitho/Winget-AutoUpdate/tree/main/Sources/WAU/Winget-AutoUpdate/mods#mods-for-wau-if-network-is-activeany-winget-is-installedrunning-as-system

```
            #Test if _WAU-mods.ps1 exist: Mods for WAU (if Network is active/any Winget is installed/running as SYSTEM)
            $Mods = "$WorkingDir\mods"
            if (Test-Path "$Mods\_WAU-mods.ps1") {
                Write-ToLog "Running Mods for WAU..." "Yellow"
                & "$Mods\_WAU-mods.ps1"
                $ModsExitCode = $LASTEXITCODE
                #If _WAU-mods.ps1 has ExitCode 1 - Re-run WAU
                if ($ModsExitCode -eq 1) {
                    Write-ToLog "Re-run WAU"
                    Start-Process powershell -ArgumentList "-NoProfile -ExecutionPolicy Bypass -Command `"$WorkingDir\winget-upgrade.ps1`""
                    Exit
                }

```

**Comment by AndrewDemski-ad-gmail-com** [comment]
Hey @ccorrion 
are you thinking about WMI call (like in Control Center) or something more sophisticated?

**Reply by ccorrion** [reply]
For what we're doing, a WMI call is enough, though it makes sense to add an extensibility point there.  If you're outputting a json summary of the run, it might make sense to clean those up to only the last 30 days worth.  I'm sure there's other things I haven't thought of yet.

**Comment by fabio-mm** [comment]
Hello, an after run mod would also be useful for me, for example to delete all desktop links created after upgrading. I tried with a sleep command on the prerun mods file but it's not forking the process so making it wait for installations too.
Otherwise i'd need to create something like 20 ps1 file one for each app, but it would still not be a complete work as there may be on a few machines a few software i'm not aware of, leaving their links on desktop again.
Any way to reach that goal or to have an afterrun Mod implemented in future versions?
thank you
regards

**Reply by AndrewDemski-ad-gmail-com** [reply]
IMHO That would be one PS1 file and 19 clones using 1st as a template.

**Reply by fabio-mm** [reply]
> IMHO That would be one PS1 file and 19 clones using 1st as a template.

not sustainable, my "20 ps1 file" was just an example to explain but we have something like 250 different apps installed on the various machines and some apps we are not aware of so we can't create a file for apps we can't forecast;

I've seen the developer was creating a simple postrun mod that would solve the problem but now it looks not implemented yet and i'm not sure what happened

**Reply by KnifMelti** [reply]
Waiting on things to stabilize regarding PR:s and the resent change in rebase.
My PR won't get trough the Linter.

**Comment by KnifMelti** [comment]
The only useful way to do this is to implement a **Mods for WAU (postsys)** - if Network is active/any Winget is installed/running as SYSTEM after SYSTEM updates - the User part of WAU haven't run (if used)
(Now done in #826) EDIT: awaiting things...

**Reply by KnifMelti** [reply]
Now merged into branch **develop** https://github.com/Romanitho/Winget-AutoUpdate/commit/d9322c89c4efbe469a50777925868971f1093d4b

**Reply by KnifMelti** [reply]
https://github.com/Romanitho/Winget-AutoUpdate/releases/tag/v2.2.12-n.0
